src/libcamera/pipeline/rpi/vc4/vc4.cpp

```diff
@@ -837,6 +837,17 @@ void Vc4CameraData::unicamBufferDequeue(FrameBuffer *buffer)
 		 * as it does not receive the FrameBuffer object.
 		 */
 		ctrl.set(controls::SensorTimestamp, buffer->metadata().timestamp);
+
+		/* Also record a wall-clock timestamp that can be passed to IPAs. */
+		while (!frameWallClock_.empty() &&
+		       frameWallClock_.front().first < buffer->metadata().sequence)
+			frameWallClock_.pop();
+
+		if (frameWallClock_.front().first == buffer->metadata().sequence) {
```

Comment:
I can't convince myself if ``frameWallClock_`` will always have an element here.  Perhaps we can be safe and add ``if (!frameWillCllock_.empty() && ...``